### PR TITLE
Touch multiselect always use toggle

### PR DESCRIPTION
**Pull request by snowystinger**
Closes 
Touch doesn't have modifier keys, so to handle multiselect, we use toggle mode so that taps can add to a selection and won't replace the current selection.

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Use on a mobile device, go to story `multi select, replace on press, select on focus` and tap multiple to get a multi select

### CODE REVIEW

File: packages/@react-aria/selection/test/useSelectableCollection.test.js [hunk 1]
```diff
@@ -52,4 +55,27 @@ describe('useSelectableCollection', () => {
     expect(options[1]).not.toHaveAttribute('aria-selected');
     expect(options[2]).toHaveAttribute('aria-selected', 'true');
   });
-});
\ No newline at end of file
+
+  it('uses toggle mode in mobile', () => {
```
Comment (snowystinger) [inline packages/@react-aria/selection/test/useSelectableCollection.test.js:59]: is it better to do it by mobile? or if we detect that it's touch driven interaction? what if they plug a keyboard into their phone?

Comment (LFDanLu) [reply packages/@react-aria/selection/test/useSelectableCollection.test.js:59]: Prob better to detect that it is touch driven interaction IMO, don't think iPads and tablets get caught by `useIsMobileDevice` (need to double check this, gotta charge my test ipad).

Comment (snowystinger) [reply packages/@react-aria/selection/test/useSelectableCollection.test.js:59]: after thinking about it some, driving this as a decision of touch does seem better
I think a touch user wouldn't reach for a keyboard modifier key

**Review by LFDanLu — APPROVED** [review]

**Comment by LFDanLu** [comment]
@snowystinger Ah just noticed that there is a conflict, I'll reapprove after

**Review by ktabors — APPROVED** [review]
LGTM

**Pull request merged**